pkg/virt-controller/network/vmcontroller_test.go

```diff
@@ -0,0 +1,328 @@
+		originalVM.Spec.Template.Spec.Networks = originalVM.Spec.Template.Spec.Networks[:1]
+		originalVM.Spec.Template.Spec.Domain.Devices.Interfaces = originalVM.Spec.Template.Spec.Domain.Devices.Interfaces[:1]
+		Expect(updatedVM).To(Equal(originalVM))
+
+		// Assert that the hotplug reached the VMI
+		updatedVMI, err := clientset.KubevirtV1().VirtualMachineInstances(vmi.Namespace).Get(context.Background(), vmi.Name, k8smetav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		Expect(updatedVM.Spec.Template.Spec.Networks).To(Equal(updatedVMI.Spec.Networks))
+		Expect(updatedVM.Spec.Template.Spec.Domain.Devices.Interfaces).To(Equal(updatedVMI.Spec.Domain.Devices.Interfaces))
+	})
+
+	It("sync succeeds to hotunplug interfaces when pod is not found", func() {
+		clientset := fake.NewSimpleClientset()
+		c := network.NewVMNetController(
+			clientset,
+			stubClusterConfig{netHotplugEnabled: true},
+			stubPodGetter{pod: nil},
+		)
+		vmi := libvmi.New(
+			libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()),
+			libvmi.WithNetwork(v1.DefaultPodNetwork()),
+			libvmi.WithInterface(libvmi.InterfaceDeviceWithBridgeBinding("foonet")),
+			libvmi.WithNetwork(libvmi.MultusNetwork("foonet", "foonet-nad")),
+		)
+		// Make sure the interfaces are visible on the status as well (otherwise hotunplug is not triggered)
+		for _, net := range vmi.Spec.Networks {
+			vmi.Status.Interfaces = append(vmi.Status.Interfaces, v1.VirtualMachineInstanceNetworkInterface{Name: net.Name})
+		}
+
+		vm := libvmi.NewVirtualMachine(vmi.DeepCopy())
+
+		// Unplug the network interface at the VM (only).
+		vm = unplugNetworkInterface(vm, "foonet")
+
+		// Simulate the existence of the VMI on the server (to allow the Sync to patch it).
+		_, err := clientset.KubevirtV1().VirtualMachineInstances(vmi.Namespace).Create(context.Background(), vmi, k8smetav1.CreateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		originalVM := vm.DeepCopy()
+		updatedVM, err := c.Sync(vm, vmi)
+		Expect(err).NotTo(HaveOccurred())
+
+		Expect(updatedVM).To(Equal(originalVM))
+
+		// Assert that the hotunplug reached the VMI
+		updatedVMI, err := clientset.KubevirtV1().VirtualMachineInstances(vmi.Namespace).Get(context.Background(), vmi.Name, k8smetav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		Expect(updatedVMI.Spec.Networks).To(Equal(updatedVM.Spec.Template.Spec.Networks))
+		Expect(updatedVMI.Spec.Domain.Devices.Interfaces).To(Equal(updatedVM.Spec.Template.Spec.Domain.Devices.Interfaces))
+	})
+
+	It("sync does not hotunplug interfaces when legacy ordinal interface names are found", func() {
+		clientset := fake.NewSimpleClientset()
+		pod := &k8sv1.Pod{
+			ObjectMeta: k8smetav1.ObjectMeta{Annotations: map[string]string{
+				networkv1.NetworkStatusAnnot: `[
+						{"interface":"eth0", "name":"red-net", "namespace": "default"},
+						{"interface":"net1", "name":"blue-net", "namespace": "default"}
```

Comment:
nit: it could be nice to have the network names equal to what is defined on the VM/VMI.

Reply:
Done

Reply:
The secondary network lacks the `-nad` suffix (since `name` represents the name of the NAD).